docs/GettingStarted.md

```diff
@@ -119,23 +141,126 @@ Check the properties of the created stream. The **format** is one property which
+It is important to close your stream when you're not using it to avoid hogging audio resources which other apps could use. To do this use:
 
     stream->close();
 
 `close()` is a blocking call which also stops the stream.
 
 It is usually advisable to close your stream when [`Activity.onPause()`](https://developer.android.com/guide/components/activities/activity-lifecycle#onpause) is called.
 
+## Using a ManagedStream
+Create and configure a builder.
+```
+oboe::AudioStreamBuilder builder;
+
+// Make sure to  always set the Performance and Sharing mode
+builder.setPerformanceMode(oboe::PerformanceMode::LowLatency)
+  ->setSharingMode(oboe::SharingMode::Exclusive); // The builder setter methods
+  can be chained for convenience
+builder.setCallback(myCallback)->setFormat(oboe::AudioFormat::Float);
+```
+Declare a ManagedStream. Make sure it is declared in an appropriate scope (e.g.
+the member of a managing class). Avoid declaring as a global.
+```
+oboe::ManagedStream managedStream;
+```
+Open the ManagedStream.
+```
+builder.openManagedStream(managedStream);
+```
+Start the ManagedStream in order to cause it to begin calling back. If
+necessary, check stream properties before requesting start (especially if some
+stream parameters were left unspecified).
+```
+managedStream->requestStart();
+```
+In order to change the configuration of the stream, simply call this method
+again. The existing stream is closed, destroyed and a new stream is built and
+populates the managedStream.
+```
+// Modify the builder with some additional properties at runtime.
+builder.setDeviceId(MY_DEVICE_ID);
+// Re-open the stream with some additional config
+// The old ManagedStream is automatically closed and deleted
+builder.openManagedStream(managedStream);
+```
+The `ManagedStream` takes care of its own closure and destruction. If used in an
+automatic allocation context (such as a member of a class), the stream does not
+need to be closed or deleted. Make sure that the object which is responsible for
+the MangedStream (its enclosing class), goes out of scope when
+`Activity.onPause()` is called.
+
+The following class is a complete implementation of a ManagedStream, which
+renders a sine wave. Creating the class (e.g. through the JNI bridge) creates
+and opens an Oboe stream which renders audio, and its destruction stops and
+closes the stream.
+```
+#include <oboe/Oboe.h>
+#include <math.h>
+
+class OboeSinePlayer: public oboe::AudioStreamCallback {
+public:
+
+
+    OboeSinePlayer() {
+        oboe::AudioStreamBuilder builder;
+        // The builder set methods can be chained for convenience.
+        builder.setSharingMode(oboe::SharingMode::Exclusive)->setPerformanceMode(oboe::PerformanceMode::LowLatency);
+        builder.setChannelCount(kChannelCount)->setSampleRate(kSampleRate)->setFormat(oboe::AudioFormat::Float);
```

Comment:
put each set() on its own line. I misread the code because of this.
You can still use continuation to chain the builder calls.